common/frame_session.go

```diff
@@ -273,6 +274,69 @@ func (fs *FrameSession) onEventBindingCalled(event *cdpruntime.EventBindingCalle
 	fs.logger.Debugf("FrameSessions:onEventBindingCalled",
 		"sid:%v tid:%v name:%s payload:%s",
 		fs.session.ID(), fs.targetID, event.Name, event.Payload)
+
+	err := fs.parseAndEmitWebVitalMetric(event.Payload)
+	if err != nil {
+		fs.logger.Errorf("FrameSession:onEventBindingCalled", "failed to emit web vital metric: %v", err)
+	}
+}
+
+func (fs *FrameSession) parseAndEmitWebVitalMetric(object string) error {
+	fs.logger.Debugf("FrameSession:parseAndEmitWebVitalMetric", "object:%s", object)
+
+	wv := struct {
+		ID             string
+		Name           string
+		Value          json.Number
+		Rating         string
+		Delta          json.Number
+		NumEntries     json.Number
+		NavigationType string
+		URL            string
+	}{}
+
+	if err := json.Unmarshal([]byte(object), &wv); err != nil {
+		return fmt.Errorf("json couldn't be parsed: '%w'", err)
+	}
+
+	metric, ok := fs.k6Metrics.WebVitals[wv.Name]
+	if !ok {
+		return fmt.Errorf("metric not registered '%v'", wv.Name)
+	}
+
+	metricRating, ok := fs.k6Metrics.WebVitals[k6ext.ConcatWebVitalNameRating(wv.Name, wv.Rating)]
+	if !ok {
+		return fmt.Errorf("metric not registered '%v'", k6ext.ConcatWebVitalNameRating(wv.Name, wv.Rating))
```

Comment:
This will wrap the string in quotes:

```suggestion
	if !ok {
		return fmt.Errorf("metric not registered %q", k6ext.ConcatWebVitalNameRating(wv.Name, wv.Rating))
```

Reply:
I added this and similar error formatting in a single commit: 2ade4fb210025e829e08786766b29fc0873f7db5